usersession/userd/privileged_desktop_launcher.go

```diff
@@ -290,12 +315,14 @@ func parseExecCommand(command string, icon string) ([]string, error) {
 			arg = arg[1:]
 		} else if strings.HasPrefix(arg, "%") {
 			switch arg {
-			case "%f", "%F", "%u", "%U":
-				// If we were launching a file with
-				// the application, these variables
-				// would expand to file names or URIs.
-				// As we're not, they are simply
-				// removed from the argument list.
+			case "%f", "%u":
+				if (arguments != nil) && (len(arguments) > 0) {
+					args = append(args, arguments[0])
+				}
+			case "%F", "%U":
+				if (arguments != nil) && (len(arguments) > 0) {
+					args = append(args, arguments...)
```

Comment:
I think this statement would be safe without the guard: a nil slice acts as an empty slice, and one argument append() is not an error.

Reply:
You are right. Tested, and works as you say.